src/components/PupilComponents/QuizResults/QuizResults.tsx

```diff
@@ -0,0 +1,146 @@
+            (questionResult, index) => {
+              const quizQuestion = quizArray[index];
+              const questionStem = quizQuestion?.questionStem;
+              const answers = quizQuestion?.answers;
+              const grade = questionResult.grade;
+              const questionType = quizQuestion?.questionType;
+              return (
+                <OakFlex
+                  key={`${lessonSection}-question-${index}`}
+                  $gap={"space-between-s"}
+                  $flexDirection={["column", "row"]}
+                  role="listitem"
+                >
+                  <OakIcon
+                    iconName={grade === 1 ? "tick" : "cross"}
+                    $background={grade === 1 ? "icon-success" : "icon-error"}
+                    $colorFilter={"white"}
+                    $borderRadius={"border-radius-circle"}
+                  />
+                  <OakFlex $flexDirection={"column"} $gap={"space-between-m"}>
+                    {questionStem && (
+                      <QuizResultQuestionStem
+                        index={index}
+                        questionStem={questionStem}
+                      />
+                    )}
+                    {answers &&
+                      answers["multiple-choice"] &&
+                      isArray(questionResult?.feedback) && (
+                        <QuizResultMCQ
+                          answers={answers["multiple-choice"]}
+                          feedback={questionResult.feedback}
+                          pupilAnswer={questionResult.pupilAnswer}
+                        />
+                      )}
+                    {answers &&
+                      answers["short-answer"] &&
+                      !isArray(questionResult?.feedback) &&
+                      questionResult?.feedback && (
+                        <QuizResultShortAnswer
+                          pupilAnswer={questionResult.pupilAnswer}
+                          feedback={questionResult.feedback}
+                        />
+                      )}
+                    {answers &&
+                      answers["order"] &&
+                      isArray(questionResult?.feedback) && (
+                        <QuizResultOrder
+                          answers={answers["order"]}
+                          feedback={questionResult.feedback}
+                          pupilAnswers={questionResult.pupilAnswer}
+                        />
+                      )}
+                    {answers &&
+                      answers["match"] &&
+                      isArray(questionResult?.feedback) && (
+                        <QuizResultMatch
+                          answers={answers["match"]}
+                          feedback={questionResult.feedback}
+                          pupilAnswers={questionResult.pupilAnswer}
+                        />
+                      )}
+                    {grade === 0 && (
```

Comment:
this nesting is difficult to read. I would again split the inner into a subcomponent